src/APL/vercelAPL.test.ts

```diff
@@ -0,0 +1,145 @@
+import { afterEach, describe, expect, it } from "vitest";
+
+import {
+  DOMAIN_VARIABLE_NAME,
+  SALEOR_DEPLOYMENT_TOKEN,
+  SALEOR_REGISTER_APP_URL,
+  TOKEN_VARIABLE_NAME,
+  VercelAPL,
+} from "./vercelAPL";
+
+describe("APL", () => {
+  describe("VercelAPL", () => {
+    describe("constructor", () => {
+      const initialEnv = { ...process.env };
+
+      afterEach(() => {
+        process.env = { ...initialEnv };
+      });
+
+      it("Raise an error when configuration is missing", async () => {
+        delete process.env[SALEOR_REGISTER_APP_URL];
+        process.env[SALEOR_DEPLOYMENT_TOKEN] = "token";
+
+        expect(() => new VercelAPL()).toThrow();
+
+        process.env[SALEOR_REGISTER_APP_URL] = "http://example.com";
+        delete process.env[SALEOR_DEPLOYMENT_TOKEN];
+
+        expect(() => new VercelAPL()).toThrow();
+      });
+    });
+
+    it("Constructor with config values", async () => {
+      expect(
+        () =>
+          new VercelAPL({
+            deploymentToken: "token",
+            registerAppURL: "http://example.com",
+          })
+      ).not.toThrow();
+    });
+
+    it("Constructor with config values from environment variables", async () => {
```

Comment:
Maybe there can be another test that constructor options will take precedence over env variables?